docs/DeveloperGuide.md

```diff
@@ -1,38 +1,331 @@
+### Architecture
+
+**Main components of the architecture**
+
+![image](./PlantUML/MainArchitecture.png)
+
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+
+Given below is a quick overview of Duke(main) components and how they interact with each other.
+It is responsible for,
+* At app launch: Initializes all other components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+The rest of the App consists of five components.
+
+* [**`UI`**](#ui-component): The UI of the App.
+* [**`Parser`**](#parser-component): Parses user input.
+* [**`Command`**](#command-component): The command executor.
+* [**`RecipeList`**](#recipelist-component): Holds the recipe data in the recipe manager by an ArrayList. 
+* [**`Storage`**](#storage-component): Reads data from, and writes data to, the hard disk.
+
+**How the architecture components interact with each other**
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+![image](./PlantUML/ArchitectureInteract.png)
+
+#### UI component
+The **API** of this component is specified in [`UI.java`](https://github.com/AY2223S2-CS2113-F13-1/tp/blob/master/src/main/java/seedu/duke/ui/UI.java)
+
+<!-- ![image](./PlantUML/UIcomponent.png) -->
+
+UI class implements the StringLib interface for some output strings. It is responsible for the following tasks:
+* Prints the welcome and goodbye messages.
+* Prints the help message, which lists all the commands available and their examples.
+* Prints the log messages for some managing operations.
+* Prints the error messages for some exceptions.
+
+#### Parser component
+The **API** of this component is specified in [`Parser.java`](https://github.com/AY2223S2-CS2113-F13-1/tp/blob/master/src/main/java/seedu/duke/parser/Parser.java)
+
+![image](./PlantUML/ParserComponent.png)
+
+Parser class implements the StringLib interface for some output strings. It is responsible for the following tasks:
+* Executes the particular command received from the user input.
+* Determines if the program termination command has been sent.
+
+#### Command component
+The **API** of this component is specified in [`Command.java`](https://github.com/AY2223S2-CS2113-F13-1/tp/blob/master/src/main/java/seedu/duke/command/Command.java)
+
+![image](./PlantUML/CommandComponent.png) 
+
+Command class the StringLib interface for some output strings. It also inherits CommandType Enums for determining the 
+correct tasks to be executed. It is responsible for the following tasks
+
+
+#### RecipeList component 
+The **API** for this component is specified in [`RecipeList.java`](https://github.com/AY2223S2-CS2113-F13-1/tp/blob/master/src/main/java/seedu/duke/recipe/RecipeList.java)
+
+![RecipeList Component](./PlantUML/RecipeListComponent.png)
+
```

Comment:
Remove the C symbols beside the Class boxes as it is not the module convention. Replace the private and public access modifiers with - and + symbols instead.